docs/spec/staking/end_block.md

```diff
@@ -1,68 +1,62 @@
-        }
-    return
-```
+Each abci end block call, the operations to update queues and validator set
+changes are specified to execute. 
 
 ## Validator Set Changes
 
-The Tendermint validator set may be updated by state transitions that run at
-the end of every block. The Tendermint validator set may be changed by
-validators either being jailed due to inactivity/unexpected behaviour (covered
-in slashing) or changed in validator power. Determining which validator set
-changes must be made occurs during staking transactions (and slashing
-transactions) - during end-block the already accounted changes are applied and
-the changes cleared
-
-```golang
-EndBlock() ValidatorSetChanges
-    vsc = GetValidTendermintUpdates()
-    ClearTendermintUpdates()
-    return vsc
-```
-
-## CompleteUnbonding
-
-Complete the unbonding and transfer the coins to the delegate. Realize any
-slashing that occurred during the unbonding period.
-
-```golang
-unbondingQueue(currTime time.Time):
-    // unbondings are in ordered queue from oldest to newest
-    for all unbondings whose CompleteTime < currTime:
-        validator = GetValidator(unbonding.ValidatorAddr)
-        AddCoins(unbonding.DelegatorAddr, unbonding.Balance)
-        removeUnbondingDelegation(unbonding)
-    return
-```
-
-## CompleteRedelegation
-
-Note that unlike CompleteUnbonding slashing of redelegating shares does not
-take place during completion. Slashing on redelegated shares takes place
-actively as a slashing occurs. The redelegation completion queue serves simply to
-clean up state, as redelegations older than an unbonding period need not be kept,
-as that is the max time that their old validator's evidence can be used to slash them.
-
-```golang
-redelegationQueue(currTime time.Time):
-    // redelegations are in ordered queue from oldest to newest
-    for all redelegations whose CompleteTime < currTime:
-        removeRedelegation(redelegation)
-    return
-```
+The staking validator set is updated during this process by state transitions
+that run at the end of every block. As a part of this process any updated
+validators are also returned back to Tendermint for inclusion in the Tendermint
+validator set which is responsible for validating Tendermint messages at the
+consensus layer. Operations are as following:
+
+ - the new validator set is taken as the top `params.MaxValidators` number of
+   validators retrieved from the ValidatorsByPower index
+ - the previous validator set is compared with the new validator set 
+   - missing validators begin unbonding
+   - new validator are instantly bonded
+
+In all cases, any validators leaving or entering the bonded validator set or
+changing balances and staying within the bonded validator set incur an update
+message which is passed back to Tendermint.
+
+## Queues 
+
+Within staking, certain state-transitions are not instantaneous but take place
+of a duration of time (typically the unbonding period). When these transitions
```

Comment:
"take place of a duration of time" is unusual - maybe "take place after a duration of time"?

Reply:
was a typo replaced with `take place over a duration of time`